src/main/java/seedu/address/logic/commands/AssignPolicyCommand.java

```diff
@@ -69,10 +70,17 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(String.format(MESSAGE_ALREADY_ASSIGNED, person.getName(), policy.getName()));
         }
 
-        Policy copyPolicy = new PolicyBuilder(policy).build();
-        Person assignedPerson = new PersonBuilder(person).addPolicies(copyPolicy).build();
+        Person assignedPerson;
+
+        if (policy.isEligible(person)) {
```

Comment:
Maybe isEligible should use toUpperCase or toLowerCase to compare the tags because if the tags are like e.g. midlife and Midlife then the person and policy wont match

Reply:
isEligible makes use of the equality method of Tags, which already uses toLowercase so case shouldn't make a difference